### ISSUE TITLE
Ambiguous dependency: Injecting interface from another module causes integration tests to fail.

### ISSUE THREAD

**Issue by JoaoCamposFrom94** [quarkusio/quarkus#6695]
**Describe the bug**
Injecting an interface on an interceptor and implementing that interface in another maven module causes the error below. By changing that interface into a concrete class the error is gone.

```java
Caused by: javax.enterprise.inject.AmbiguousResolutionException:
Ambiguous dependencies for type utils.ReferenceErrorCode and qualifiers [@Default]
	- java member: utils.CustomInterceptor#<init>()
	- declared on INTERCEPTOR bean [bindings=[@Annotation], target=Optional[utils.CustomInterceptor]]
	- available beans:
		- CLASS bean [types=[utils.ReferenceErrorCode, java.lang.Object, tck.utils.ReferenceErrorCodeProvider], qualifiers=[@Default, @Any], target=tck.utils.ReferenceErrorCodeProvider]
		- CLASS bean [types=[utils.ReferenceErrorCode, java.lang.Object, tck.utils.ReferenceErrorCodeProvider], qualifiers=[@Default, @Any], target=tck.utils.ReferenceErrorCodeProvider]
```

**Expected behavior**
The integration test should pass.

**To Reproduce**
1. Clone https://github.com/talkdesk-tdx/quarkus-bug-report/tree/6a5297fbe7d0a414cd1ac6ca5b13f9d1c829605b
2. Run 'mvn install' on the root directory of the project.

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: Darwin 208.243.108.93.rev.vodafone.pt 18.7.0 Darwin Kernel Version 18.7.0: Tue Aug 20 16:57:14 PDT 2019; root:xnu-4903.271.2~2/RELEASE_X86_64 x86_64
 - Output of `java -version`: openjdk version "11.0.4" 2019-07-16
 - Quarkus version or git rev: 1.1.1.Final

**Comment by geoand** [comment]
@manovotn or @mkouba , care to take a look?

**Comment by gsmet** [comment]
@stuartwdouglas could it be related to class loading?

**Comment by mkouba** [comment]
Hm, one thing I've noticed is that you're using failsafe instead of surefire to run the test. If I change the name of the test to `ConstraintViolationExceptionMapperTest` I get a different error when running `mvn clean test`:
```
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 2.398 s <<< FAILURE! - in org.example.errorhandling.ConstraintViolationExceptionMapperTest
[ERROR] testResponse  Time elapsed: 0.047 s  <<< ERROR!
java.lang.RuntimeException: java.lang.ClassNotFoundException: org.glassfish.jersey.internal.RuntimeDelegateImpl
	at org.example.errorhandling.ConstraintViolationExceptionMapperTest.testResponse(ConstraintViolationExceptionMapperTest.java:35)
Caused by: java.lang.ClassNotFoundException: org.glassfish.jersey.internal.RuntimeDelegateImpl
	at org.example.errorhandling.ConstraintViolationExceptionMapperTest.testResponse(ConstraintViolationExceptionMapperTest.java:35)
```

**Comment by mkouba** [comment]
@gsmet I don't think Quarkus supports failsafe to be used to run `@QuarkusTest` tests ^, or?

**Comment by stuartwdouglas** [comment]
This is due to the combination of failsafe + jandex plugin, see https://groups.google.com/d/msg/quarkus-dev/uE8uWciCaRA/eoihkq3_BwAJ

**Comment by JoaoCamposFrom94** [comment]
> Hm, one thing I've noticed is that you're using failsafe instead of surefire to run the test. If I change the name of the test to `ConstraintViolationExceptionMapperTest` I get a different error when running `mvn clean test`:
> 
> ```
> [ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 2.398 s <<< FAILURE! - in org.example.errorhandling.ConstraintViolationExceptionMapperTest
> [ERROR] testResponse  Time elapsed: 0.047 s  <<< ERROR!
> java.lang.RuntimeException: java.lang.ClassNotFoundException: org.glassfish.jersey.internal.RuntimeDelegateImpl
> 	at org.example.errorhandling.ConstraintViolationExceptionMapperTest.testResponse(ConstraintViolationExceptionMapperTest.java:35)
> Caused by: java.lang.ClassNotFoundException: org.glassfish.jersey.internal.RuntimeDelegateImpl
> 	at org.example.errorhandling.ConstraintViolationExceptionMapperTest.testResponse(ConstraintViolationExceptionMapperTest.java:35)
> ```

Fixed that, I seems I stripped too many dependencies from the original code. I've added quarkus-resteasy-jsonb. You may git pull the latest version of the repo.

**Comment by JoaoCamposFrom94** [comment]
I got it fixed on the original multi-module maven project.
When I removed the jandex plugin from the library modules I got an unsatisfied exception. To fix this I removed the jandex plugin from the parent pom and moved it to each module, with the **exception of the tests**.

This way I can run unit tests and integration tests. I'll update the repository (with code and a README file) used in the bug description to include this fix.
